src/main/java/edu/illinois/cs/dt/tools/detection/DetectorPlugin.java

```diff
@@ -258,21 +271,40 @@ private Void detectorExecute(final ErrorLogger logger, final MavenProject mavenP
 
                     final List<String> tests = new ArrayList<>();
 
+                    String delimiter = testFramework.getDelimiter();
+
                     for (final TestClassData classData : testClassData) {
                         for (final String testName : classData.testNames) {
-                            tests.add(classData.className + "." + testName);
+                            tests.add(classData.className + delimiter + testName);
                         }
                     }
 
                     return tests;
                 } else {
-                    return JavaConverters.bufferAsJavaList(TestLocator.tests(mavenProject).toBuffer());
+                    return JavaConverters.bufferAsJavaList(
+                            TestLocator.tests(mavenProject, testFramework).toBuffer());
                 }
             } catch (Exception ignored) {}
 
-            return JavaConverters.bufferAsJavaList(TestLocator.tests(mavenProject).toBuffer());
+            return JavaConverters.bufferAsJavaList(TestLocator.tests(mavenProject, testFramework)
+                    .toBuffer());
         } else {
             return Files.readAllLines(DetectorPathManager.originalOrderPath());
         }
     }
+
+    private static List<Runner> removeZombieRunners(
+            List<Runner> runners, MavenProject mavenProject) throws IOException {
+        // Some projects may include test frameworks without corresponding tests.
+        // Filter out such zombie test frameworks (runners).
+        // For example, a project have both JUnit 4 and 5 dependencies, but there is
+        // no JUnit 4 tests. In such case, we need to remove the JUnit 4 runner.
+        List<Runner> aliveRunners = new ArrayList<>();
+        for (Runner runner : runners) {
+            if (getOriginalOrder(mavenProject, runner.framework(), true).size() > 0) {
```

Comment:
If the original-order file is already there or even if the conditions are in place for parsing the mvn-test.log and Surefire reports, is this a meaningful way to check for "zombie" runners? Is it better to somehow force detection of test files to see if they're JUnit 4 or JUnit 5?

Reply:
I'm not very sure about your point here. What I do here is to do the force detection. The third parameter of the method `getOriginalOrder` is set to `true`, which means it will ignore the existing original-order file.

Reply:
You're right, sorry I missed the flag.

Related, there is a potential other issue with how `getOriginalOrder` would try to parse any existing mvn-test.log and Surefire reports to get at the original order instead of using the `TestLocator` logic from testrunner. If it goes along this path, would there be a problem in trying to guess at what the proper framework to use is? Is there any information from parsing those files as to guess at which JUnit was run?

Reply:
You're right, that could be a problem. Let me use the TestLocator directly.

Reply:
Done